### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by jfcg**
Add Github CodeQL security checks for Go.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'javascript', 'python' ]
```
Comment (magik6k) [inline .github/workflows/codeql-analysis.yml:31]: We can drop python / js as those languages aren't used for anything beyond some dev scripts / tools

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '37 14 * * 5'
```
Comment (magik6k) [inline .github/workflows/codeql-analysis.yml:21]: Does it make sense to run this periodically if we already run this on all PR / master commits?

**Review by magik6k — COMMENTED** [review]
Thanks for the PR.

This looks good, but I'd like to see what we can get from this, and after doing a couple of searches, only things I find is that this gives us "security checks" by analyzing the code, and makes it possible to run quires against the code.

Are there any concrete examples of what security checks this does, and how are they made visible to us? (just want to understand how this is useful to us before merging)

**Comment by jfcg** [comment]
In my fork under `/security/code-scanning` I see:
lotus

You can see the alerts of the merge [here](https://github.com/filecoin-project/lotus/pull/6021/checks?check_run_id=2327208253)

**Review by magik6k — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**